pulp_container/app/access_policy.py

```diff
@@ -2,167 +2,14 @@
-            return True
-        else:
-            return request.user.has_perm(permission) or request.user.has_perm(
-                permission, view.get_object()
-            )
-
-    def obj_exists(self, request, view, action):
-        """
-        Check if the distribution exists.
-        """
-        return view.get_object() is not None
-
-    def is_private(self, request, view, action):
-        """
-        Check if the distribution is marked private.
-        """
-        return view.get_object().private
-
-    def namespace_is_username(self, request, view, action):
-        """
-        Check if the namespace in the request matches the username.
-        """
-        base_path = request.data.get("base_path")
-        if not base_path:
-            return False
-        namespace = base_path.split("/")[0]
-        return namespace == request.user.username
-
-    def has_namespace_model_perms(self, request, view, action):
-        """
-        Check if the user can create namespaces.
-        """
-        if request.user.has_perm("container.add_containernamespace"):
-            return True
-        return False
-
-
-class NamespacedAccessPolicyFromDB(AccessPolicyFromDB, NamespacedAccessPolicyMixin):
-    """
-    Access policy for ContainerDistributionViewSet which handles
-    namespace permissions.
-    """
-
-
-class PushRepositoryAccessPolicy(NamespacedAccessPolicyFromDB):
-    """
-    Access policy for ContainerPushRepositoryViewSet.
-    """
-
-    def has_distribution_perms(self, request, view, action, permission):
-        """
-        Check if the user has permissions on the corresponding distribution.
-        Model or object permission is sufficient.
-        """
-        if request.user.has_perm(permission):
-            return True
-        distributions = view.get_object().distributions.all()
-        return any(
-            (
-                request.user.has_perm(permission, distribution.cast())
-                for distribution in distributions
-            )
-        )
-
-
-class PushRepositoryVersionAccessPolicy(NamespacedAccessPolicyFromDB):
-    """
-    Access policy for ContainerPushRepositoryVersionViewSet.
-    """
-
-    def has_distribution_perms(self, request, view, action, permission):
-        """
-        Check if the user has permissions on the corresponding distribution.
-        Model or object permission is sufficient.
-        """
-        if request.user.has_perm(permission):
-            return True
-        repository = Repository.objects.get(pk=view.kwargs["repository_pk"])
-        distributions = repository.distributions.all()
-        return any(
-            (
-                request.user.has_perm(permission, distribution.cast())
-                for distribution in distributions
-            )
-        )
-
-
-class NamespaceAccessPolicy(AccessPolicyFromDB):
-    """
-    Access policy for ContainerNamespaceViewSet.
-    """
-
-    def namespace_is_username(self, request, view, action):
```

Comment:
we at least will need to re-write namespace_is_username and has_distribution_perms because we have multiple implementations of those depending which viewset it is meant for.

Reply:
you already did that :+1:

Reply:
Yes. Flake is rather picky with multiply defined functions.